core/src/main/scala/pureconfig/ConvertHelpers.scala

```diff
@@ -32,6 +32,9 @@ trait ConvertHelpers {
         ConfigReaderFailures(headImproved, tailImproved)
     }
 
+  private[pureconfig] def toResult[A, B](f: A => B): A => Option[ConfigValueLocation] => Either[ConfigReaderFailures, B] =
```

Comment:
We should probably do a cleanup of these private methods in pureconfig. I have not looked at them throughly, but I believe that we can simplify these utility methods to a few core ones. For this PR this is fine though, let's leave it for another time :)